### PR TITLE
bugfix & improvements

### PR DESCRIPTION
**Pull request by alexalouit**
add semaphore locker
add cli timeout
add db unique key

### CODE REVIEW

File: cliupdate.php [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <?php
+set_time_limit(300);
```
Comment (jtojnar) [inline cliupdate.php]: Why this time limit? We are already using a greater one in the `ContentLoader`.

**Comment by alexalouit** [comment]
```SOURCE_UPDATE_MAX_INTERVAL``` fix in 4726c4c29cbc8bd4393f2d58f02c9c60734f4543

File: daos/sqlite/Database.php [hunk 2]
```diff
@@ -234,6 +234,47 @@ public function __construct() {
                         'INSERT INTO version (version) VALUES (11)'
                     ]);
                 }
+                if (strnatcmp($version, '12') < 0) {
+                    \F3::get('db')->exec([
+                        // Table needs to be re-created because ALTER TABLE is rather limited.
```
[the remark this answers is not in the document]
Comment (alexalouit) [reply daos/sqlite/Database.php]: It is better? 2cf87cad222a46cbfe9756869f5ef2f5bee9b2f8

File: cliupdate.php (hunk 1 above)
Comment (alexalouit) [reply cliupdate.php]: fix in 54235f84fd374354beb3e80e68d3ecca0590206d

**Comment by alexalouit** [comment]
query fixed in 54235f84fd374354beb3e80e68d3ecca0590206d

**Comment by jtojnar** [comment]
Regarding the locking, in https://github.com/SSilence/selfoss/issues/967#issuecomment-319159139, I actually meant the `flock` command as demonstrated in https://serverfault.com/questions/82857/prevent-duplicate-cron-jobs-running. On selfoss site it would mean creating `cliupdate_flock.php` that would allow only one instance of itself at a time.

**Comment by jtojnar** [comment]
Also it would be nice if you could split the unique constraint to a separate commit from the locking part.

**Comment by alexalouit** [comment]
see #994 and #995

**Pull request closed without merging**